kolibri/core/analytics/utils.py

```diff
@@ -62,6 +73,133 @@
 ]
 
 
+def variance(data):
+    if data:
+        # calculate mean
+        m = sum(data) / len(data)
+        # return variance
+        return sum((d - m) ** 2 for d in data) / len(data)
+    return None
+
+
+def calculate_demographic_stats(dataset_id=None, channel_id=None, learner=True):
+    stats = {}
+    roles_filter = Q(roles__isnull=False)
+    queryset = FacilityUser.objects.all()
+    if learner:
+        roles_filter = Q(roles__isnull=True)
+    # handle stats at facility level
+    if dataset_id:
+        queryset = queryset.filter(roles_filter, dataset_id=dataset_id)
+    # handle stats at channel level
+    if channel_id:
+        queryset = queryset.filter(
+            roles_filter, contentsummarylog__channel_id=channel_id
+        ).distinct()
+
+    # calculate stats if there are USER_THRESHOLD users or more
+    if queryset.count() >= USER_THRESHOLD:
+        # get list of all birth years to calculate variance
+        list_of_birth_years = (
+            queryset.exclude(
+                Q(birth_year="")
+                | Q(birth_year=demographics.NOT_SPECIFIED)
+                | Q(birth_year=demographics.DEFERRED)
+            )
+            # convert birth year to integer
+            .annotate(
+                birth_year_int=Cast("birth_year", output_field=IntegerField())
+            ).values_list("birth_year_int", flat=True)
+        )
+
+        demographic_stats = (
+            queryset
+            # convert birth year to integer
+            .annotate(
+                birth_year_int=Cast("birth_year", output_field=IntegerField())
+            ).aggregate(
```

Comment:
In the future, we should update this to loop through all the genders specified in the queryset in order to generate an aggregate for each, in case we update the interface to allow for gender self selection.